File: app/backend/java/auth-service/src/main/java/com/srbms/authservice/entity/Role.java
package com.srbms.authservice.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "roles")
public class Role {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "role_id")
    private Integer roleId;

    @Column(name = "role_value", nullable = false, unique = true)
    private String roleValue;

    public Role() {}

    public Role(String roleValue) {
        this.roleValue = roleValue;
    }

    public Role(Integer roleId, String roleValue) {
        this.roleId = roleId;
        this.roleValue = roleValue;
    }

    public Integer getRoleId() {
        return roleId;
    }

    public void setRoleId(Integer roleId) {
        this.roleId = roleId;
    }

    public String getRoleValue() {
        return roleValue;
    }

    public void setRoleValue(String roleValue) {
        this.roleValue = roleValue;
    }
}

File: app/backend/.net/SmartRide.AuthService/Models/Entities/Role.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartRide.AuthService.Models.Entities;

[Table("roles")]
public class Role
{
    [Key]
    [Column("role_id")]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int RoleId { get; set; }

    [Required]
    [Column("role_value")]
    [StringLength(50)]
    public string RoleValue { get; set; } = string.Empty;

    public Role() { }

    public Role(string roleValue)
    {
        RoleValue = roleValue;
    }

    public Role(int roleId, string roleValue)
    {
        RoleId = roleId;
        RoleValue = roleValue;
    }
}
